neps/nep-0452.md

```diff
@@ -0,0 +1,164 @@
+/// (either new or existing) when a key is successfully used.
+type FTData = {
+   /// The number of tokens to transfer, wrapped in quotes and treated
+   /// like a string, although the number will be stored as an unsigned integer
+   /// with 128 bits.
+   amount: string,
+   /// The valid NEAR account indicating the Fungible Token contract.
+   contract_id: string
+}
+
+
+
+
+/****************/
+/* VIEW METHODS */
+/****************/
+
+// Returns the $yoctoNEAR amount associated with a given public key 
+// Panics if the key does not exist 
+function get_key_balance(key: PublicKey) -> string
+
+// Returns the KeyInfo associated with a given public key
+// Panics if the key does not exist
+function get_key_information(key: PublicKey) -> KeyInfo
+
+/******************/
+/* CHANGE METHODS */
+/******************/
+
+// Transfer all assets linked to the signer’s public key to an *existing* NEAR account. 
+// If the transfer fails for whatever reason, it is up to the smart contract developer to 
+// choose what should happen. For example, the contract can choose to keep the assets 
+// or send them back to the original linkdrop creator.
+// 
+// Requirements:
+// * The predecessor account *MUST* be the current contract ID.
+// * The assets being sent *MUST* be associated with the signer’s public key.
+// * The assets *MUST* be sent to the `account_id` passed in.
+//
+// Arguments:
+// * `account_id` the account that should receive the linkdrop assets.
+function claim(account_id: string) -> Promise
+
+
+
+// Creates a new NEAR account and transfers all assets linked to the signer’s public key to 
+// the *newly created account*. If the transfer fails for whatever reason, it is up to the 
+// smart contract developer to choose what should happen. For example, the contract can    
+// choose to keep the assets or return them to the original linkdrop creator. 
+//
+// Requirements 
+// * The predecessor account *MUST* be the current contract ID.
+// * The assets being sent *MUST* be associated with the signer’s public key.
+// * The assets *MUST* be sent to the `new_account_id` passed in.
+// * The newly created account *MUST* have a new access key added to its account (either       
+//   full or limited access) in the same receipt that the account was created in. 
+// 
+// Arguments
+// * `new_account_id`: the valid NEAR account which is being created and should 
+//   receive the linkdrop assets
+// * `new_public_key`: the valid public key that should be used for the access key added to         
+//   the newly created account.
+function create_account_and_claim(new_account_id: string, new_public_key: string) -> Promise
+```
+
+## Reference Implementation
+
+Below are some references for linkdrop contracts. A specific implementation of this standard can be seen in [this pull request](https://github.com/near/near-linkdrop/pull/24).◊
+
+- [Link Drop Contract](https://github.com/near/near-linkdrop)
```

Comment:
One more: https://neardrop.io/ (https://github.com/agileurbanite/contracts.near-linkdrop)

Reply:
I'd rather not include this contract as it has some pretty serious security vulnerabilities.